stats.go

```diff
@@ -15,13 +16,15 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/rcrowley/go-metrics"
 	"github.com/sirupsen/logrus"
+	"github.com/slackhq/nebula/cert"
 	"github.com/slackhq/nebula/config"
 )
 
 // startStats initializes stats from config. On success, if any futher work
 // is needed to serve stats, it returns a func to handle that work. If no
 // work is needed, it'll return nil. On failure, it returns nil, error.
-func startStats(l *logrus.Logger, c *config.C, buildVersion string, configTest bool) (func(), error) {
+func startStats(l *logrus.Logger, c *config.C, buildVersion string, clientCert *CertState, caCert *cert.NebulaCAPool, configTest bool) (func(), error) {
```

Comment:
We will lose the references to `clientCert` and `caCert` on reload as it stands. We should refactor to use `f.emitStats` instead https://github.com/slackhq/nebula/blob/b5b9d33ee753b510b316f15b503ecb11209fdf9b/interface.go#L309

Using the `Interface`s reference to `f.certState` and `f.caPool`.

We also support both prometheus and graphite. I think many of these stats would work well in a graphite world where labels aren't readily supported, so we would need to either make these work as best as possible with `gometrics` or teach `Interface` to know which stats system is being used and export the appropriate information for graphite or prometheus.